__tests__/app.test.js

```diff
@@ -162,4 +153,59 @@ describe("GET /api/articles/:article_id/comments", () => {
             expect(body.msg).toBe("Bad Request")
         })
     })
+})
+
+describe("POST /api/articles/:article_id/comments", () => {
+    it("Should return a status code: 201 and the posted comment", () => {
+        const newComment = { username: 'butter_bridge', body: 'Im not Kakarot?'}
+        
+        return request(app)
+             .post("/api/articles/1/comments")
+             .send(newComment)
+             .expect(201)
+             .then(({ body }) => {
+                const { comment } = body
+                
+                expect(comment).toEqual({
+                    comment_id: expect.any(Number),
+                    body: expect.any(String),
+                    votes: expect.any(Number),
+                    author: expect.any(String),
+                    article_id: expect.any(Number),
+                    created_at: expect.any(String)
+                })
+             })
+    })
+    it("Should return a status code: 400 when articles_id is NaN", () => {
+        const newComment = { username: 'butter_bridge', body: 'Im not Kakarot?'}
+        
+        return request(app)
+             .post("/api/articles/seven/comments")
+             .send(newComment)
+             .expect(400)
+             .then(({ body }) => {
+                expect(body.msg).toBe("Bad Request")
+             })
+    })
+    it("Should return a status code: 404 when articles_id doesn't exist", () => {
+        const newComment = { username: 'butter_bridge', body: 'Im not Kakarot?'}
+        
+        return request(app)
+             .post("/api/articles/420/comments")
+             .send(newComment)
+             .expect(404)
+             .then(({ body }) => {
+                expect(body.msg).toBe("Not Found")
+             })
+    })
+    it("Should return error 422 when using invalid input (username/body isnt present/valid)", ()=> {
+        const invalidComment = { username: 'butter_bridge' }
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send(invalidComment)
+        .expect(422)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Unprocessable Entity")
+        })
+    })
```

Comment:
Nice error handling, but one missing test here is when the username is not a valid username i.e. `{ username: 'invalidUsername', body: 'hello!!'}`